src/dmd/astbase.d

```diff
@@ -1441,6 +1441,21 @@ struct ASTBase
             srcfile = new File(srcfilename);
         }
 
+        /**
+        A root module is one that will be compiled to object code.
+        */
+        final bool isRoot() const
+        {
+            return true;
```

Comment:
When does isRoot return false for a module such that its unittests don't compile?

Reply:
isRoot will return false if the module isn't "taken" to object code (see comment).  This will happen if the module is not passed on the command line and is not included as a "compiled import"

Reply:
I guess I'm just asking because this looks pretty constant (final, returns true). What is the type of AST class that represents an import module?

Reply:
OK, I'm seeing that this is duplicated in `dmodule.d`, but with a different implementation (one that makes sense). I'm not sure what the `astbase.d` file does, but it sure seems like it's doing something different than `dmodule.d`. What is this for?

Reply:
Was just looking through the comments and saw that your question wasn't answered.  The module class you are looking at is missing ALOT of information, it's just as STUB as far as I know probably meant to hold a future refactored version of the current `Module` class in `dmodule.d`.

Reply:
I had already figured it out. it seems it's there for when you want to instantiate the parser only, and not actually compile anything. Perhaps if you aren't generating object code.